src/httprpc.cpp

```diff
@@ -319,3 +319,8 @@ void StopHTTPRPC()
         httpRPCTimerInterface.reset();
     }
 }
+
+const std::set<std::string>& GetWhitelistedRpcs(const std::string& user_name)
+{
+    return g_rpc_whitelist.at(user_name);
```

Comment:
This fails when user is not in the map.

Reply:
>Not clear what it should do for users that don't have a whitelist (ie, can access all methods). Currently errors I think.

Reply:
I believe it should return the list of all available RPC methods. It's consistent and it'd enable feature detection. See #19087 for more discussion about feature detection.

Reply:
If the rpc whitelist is empty, one could call `help` to get all methods. But maybe just returning the whole set here seems appropriate. How to deal with hidden RPCs, though?

Reply:
I think it should return empty - no whitelist is defined for the user.

Reply:
Hmm, should it be designed for simplicity of `bitcoind` or practicality of clients? I suppose this feature will be most useful for auto-detecting available features, but maybe I'm missing something?

Reply:
I guess that makes sense, but I think hidden methods should be turned off by default.

Reply:
Do I understand correctly that the hidden methods are used for testing only?